src/BinSkim.Rules/DoNotShipVulnerableBinaries.cs

```diff
@@ -80,13 +80,22 @@ private static StringToVersionMap BuildDefaultVulnerableBinariesMap()
 
         public override void Analyze(BinaryAnalyzerContext context)
         {
+            // Throw a platform unsupported exception--FileVersionInfo does not behave "correctly" on Linux.
+            BinaryParsers.PlatformSpecificHelpers.ThrowIfNotOnWindows();
+            
             string fileName = Path.GetFileName(context.PE.FileName);
 
             Version minimumVersion;
             if (context.Policy.GetProperty(VulnerableBinaries).TryGetValue(fileName, out minimumVersion))
             {
                 FileVersionInfo fvi = FileVersionInfo.GetVersionInfo(Path.GetFullPath(context.PE.FileName));
                 string rawVersion = fvi.FileVersion;
+                // Version is sometimes null on non-Windows platforms.

```

Comment:
You can replace this code by making the previous line look like so:

string rawVersion = fvi.FileVersion ?? String.Empty;